### PR TITLE
refactor!: change key format from camel case or snake case to lower hyphen 

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description

Make indicator and report keys as well as API parameters consistent by changing the format from camel case or snake case to lower hyphen.

Throughout the code base indicator and report keys where changed from camel case (`MappingSaturation`) to lower hyphen (`mapping-saturation`).

API parameter and variable names  where changed from lower camel case (`layerKey`) to lower hyphen (`layer-key`)  in the case of API paramter or snake-case (`layer_key`) in the case of variable names.

Layer keys have not been changed from snake case to hyphen case.

### Corresponding issue

Related issues: #485 #50

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)
- [x] fix tests
- [x] remove unused helper functions
- [x] update website

### CODE REVIEW

File: workers/tests/integrationtests/test_api_report.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def test_indicator_dataset_invalid(self):
         parameters = {
             "name": self.report_name,
             "dataset": "foo",
-            "featureId": self.feature_id,
+            "feature-id": self.feature_id,
```
Comment (joker234) [inline workers/tests/integrationtests/test_api_report.py:133]:
```suggestion
            "feature_id": self.feature_id,
```

File: workers/tests/integrationtests/test_api_report.py [hunk 2]
```diff
@@ -149,7 +149,7 @@ def test_indicator_invalid_set_of_arguments(self):
             },
             {
                 "name": self.report_name,
-                "feature_id": "3",
+                "feature-id": "3",
```
Comment (joker234) [inline workers/tests/integrationtests/test_api_report.py:152]:
```suggestion
                "feature_id": "3",
```

File: workers/tests/integrationtests/test_api_report_geojson_io.py [hunk 3]
```diff
@@ -88,7 +88,7 @@ def test_invalid_set_of_arguments(self):
             "name": self.report_name,
             "bpolys": self.feature,
             "dataset": "foo",
-            "featureId": "3",
+            "feature-id": "3",
```
Comment (joker234) [inline workers/tests/integrationtests/test_api_report_geojson_io.py:91]:
```suggestion
            "feature_id": "3",
```

File: workers/tests/unittests/test_helper.py [hunk 4]
```diff
@@ -41,21 +44,15 @@ def run_tests(self, raw: dict, number_of_features: int = 1) -> None:
...
-            name_to_class(class_type="indicator", name="Minimal"),
+            name_to_class(class_type="indicator", name="minimal"),
             MinimalIndicator,
         )
 
         self.assertIs(
-            name_to_class(class_type="report", name="Minimal"),
+            name_to_class(class_type="report", name="minimal"),
             MinimalReport,
         )
```
Comment (joker234) [inline workers/tests/unittests/test_helper.py:55]: Re-add this?
```suggestion

        self.indicators = load_metadata("indicators")
        for indicator_name in self.indicators.keys():
            self.assertIsNotNone(
                name_to_class(class_type="indicator", name=indicator_name)
            )

```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,11 @@
 
 ## Current Main
 
+### Breaking Changes
+
+- refactor!: change parameter names from snake case to lower hyphen ([#398])
+- refactor!: change indicator and report keys from lower camel case to lower hyphen ([#398])
```
Comment (joker234) [inline CHANGELOG.md:8]:
```suggestion
- change parameter names from snake case to lower hyphen ([#398])
- change indicator and report keys from lower camel case to lower hyphen ([#398])
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,7 +15,21 @@
...
+- rename API request parameters from camel case to lower hyphen ([#398])
+
+| old           | new            |
+| ---           | ---            |
+| `layerKey`    | `layer-key`    |
+| `includeSvg`  | `include-svg`  |
+| `includeHtml` | `include-html` |
+| `featureId`   | `feature-id`   |
+| `fidField`    | `indicators`   |
+
```
Comment (joker234) [inline CHANGELOG.md:30]: I think this is only true for `POST`, isn't it? Maybe make that clear here?

**Review by joker234 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (matthiasschaub) [reply CHANGELOG.md:8]: apllied

**Review by joker234 — APPROVED** [review]

**Review by joker234 — APPROVED** [review]

**Pull request merged**